draco/debug.py

```diff
@@ -0,0 +1,218 @@
+from functools import cached_property
+from typing import Iterable, NamedTuple
+
+import altair as alt
+import pandas as pd
+
+from .draco import Draco
+
+
+class DracoDebug:
+    __DEFAULT_DRACO__ = Draco()
+    __DEFAULT_FEATURE_NAMES__ = set(__DEFAULT_DRACO__.soft_constraint_names)
+    __DEFAULT_WEIGHTS__ = __DEFAULT_DRACO__.weights
+
+    def __init__(
+        self,
+        specs: dict[str, Iterable[str] | str],
+        draco: Draco | None = None,
+        feature_names: set[str] | None = None,
+        weights: dict[str, int] | None = None,
+    ):
+        """
+        Initializes a new debugging instance.
+
+        :param specs: ``dict`` of specifications to analyze, keys representing
+                       the spec name and values the corresponding ASP declarations.
+        :param draco: The ``Draco`` instance to be debugged
+        :param feature_names: A set of the constraints' names which should be debugged
+        :param weights: A ``dict`` associating weights with feature (constraint) names.
+                        Keys of this ``dict`` are expected to be of the form
+                        ``"{feature_name}_weight"``. If the supplied ``feature_names``
+                        set contains a feature name for which no entry is present in
+                        this ``dict``, the default weight of the feature will be used.
+        """
+        self.specs = specs
+        cls = DracoDebug
+        if feature_names is None:
+            feature_names = cls.__DEFAULT_FEATURE_NAMES__
```

Comment:
I think this could be a bug. We don't use the provided Draco instance so the features may be different.

Reply:
That's right, thank you for spotting. I addressed this in dabb05b by removing the option to enter custom weights and feature names in the `DracoDebug.__init__`, as it would just lead to confusing, inconsistent behaviour. Users will have the chance to configure the debugger just by dependency injecting a pre-configured `Draco` instance along with the specs.